enterprise/api/v1/views.py

```diff
@@ -213,6 +213,110 @@ def course_enrollments(self, request, pk):
 
         return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
 
+    @action(detail=True, methods=['post'], permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_enroll_learners', fn=lambda request, pk: pk)
+    # pylint: disable=invalid-name,unused-argument
+    def enroll_learners_in_courses(self, request, pk):
+        """
+        Creates a set of licensed enterprise_learners by bulk enrolling them in the specified course.
+
+        Note: This endpoint isn't transactional in that if an enrollment fails, there can potentially be a scenario
+        with a subset of users enrolled. Requested learners' eligibility to enroll are expected to be pre-vetted.
+
+        Expected params:
+            - license_info (Dict): user emails as keys and the license UUIDs associated with each of the courses the
+            user's being enrolled in as values. Note, each user is expected to have exactly one license per course
+            listed in the 'courses' parameter.
+            Example:
+                license_info: {
+                    'newuser@test.com': {
+                        'course-v1:edX+DemoX+Demo_Course': '5b77bdbade7b4fcb838f8111b68e18ae'
+                    },
+                    ...
+                }
+
+            - courses (Dict): course run keys as keys and the course mode as the value.
+            Example:
+                'courses': {
+                    'course-v1:edX+DemoX+Demo_Course': 'verified',
+                    ...
+                }
+        """
+        enterprise_customer = self.get_object()
+        serializer = serializers.EnterpriseCustomerBulkSubscriptionEnrollmentsSerializer(
+            data=request.data,
+            context={
+                'enterprise_customer': enterprise_customer,
+                'request_user': request.user,
+            }
+        )
+        if serializer.is_valid(raise_exception=True):
+            errors = []
+            enrolled_count = 0
+            license_uuids = serializer.validated_data.get('license_info')
+            course_run_keys = serializer.validated_data.get('courses')
+            users_missing_licenses = []
+
+            # Get a list of user emails from the license info.
+            emails = set(license_uuids.keys())
+            try:
+                for email in emails:
+                    try:
+                        validate_email_to_link(email, enterprise_customer, raise_exception=False)
+                        if len(license_uuids[email]) != len(course_run_keys):
+                            users_missing_licenses.append(email)
+                    except ValidationError as error:
+                        errors.append(error)
+            except ValidationError as exc:
+                errors.append(exc)
+
+            if users_missing_licenses:
```

Comment:
Do our errors not require a standard return format? Can you use a string for one and a list for the other?

Reply:
A standardized format would be nice, but I haven't come across a required one. I'm down to discuss stricter behavioral patterns if you wish. Any suggestions?

Reply:
Ugh, another item for the architecture meeting.  I think part of my problems are happening because we normally try to have validation errors in the serializer but this is validating semantic content, not formatting problems. However, in an effort to be consistent with what the rest of the Django setup is doing, it looks returning a list of errors matches the Django REST pattern in use elsewhere.